File: blockchain.js

const uuid = require('uuid/v1')


function Blockchain() {
  this.currentNode = `http://localhost:${parseInt(process.env.PORT)}`
  this.nodes = []
  this.pendingTransactions =  []
}

Blockchain.prototype.createNewTransaction = function(sender, recipient, value) {
  const transaction = {
    sender, recipient, value, transactionId: uuid()
  }
  return transaction
}

Blockchain.prototype.addToPendingTransaction = function(transaction) {
  this.pendingTransactions.push(transaction)

  return this.pendingTransactions.length
}

module.exports = Blockchain

File: blockchain.py

import sys
from urllib.parse import urlparse

class Blockchain(object):
    def __init__(self):
        self.chain = []
        self.nodes = set()
        self.currentNode = f'http://localhost:{sys.argv[1]}'
        self.pendingTransactions = []
    
    def registerNode(self, address):
        self.nodes.add(address)